.environment

```diff
@@ -9,3 +9,6 @@ if [ -n "$PLATFORM_APP_DIR" -a -f "$PLATFORM_APP_DIR"/composer.json ] ; then
   bin=$(composer config bin-dir --working-dir="$PLATFORM_APP_DIR" --no-interaction 2>/dev/null)
   export PATH="${PLATFORM_APP_DIR}/${bin:-vendor/bin}:${PATH}"
 fi
+
+# Set the default site URL for drush.
+export DRUSH_OPTIONS_URI=$(echo $PLATFORM_ROUTES | base64 --decode | jq -r 'map(select(.primary == true)) | first | .production_url')
```

Comment:
I have a couple questions: 

1) is `production_url` always set, and if not, doesn't that mean that `DRUSH_OPTIONS_URI` will be an empty string rather than undefined... and I wonder how drush would play with that environment variable being set but empty.

2) How would this affect environments that are not main/master? Those URLs will most likely have no resemblance to the production url, and therefore would not allow drush to derive which multi-site to load.

Reply:
Thanks @apotek . After looking closer I think you are correct that I shouldn't be using the `production_url`... the `production_url` does seem to always be present, but is always the URL of the "default" branch.

For example, here is the `$PLATFORM_ROUTES` environment on a `test` branch. See that `production_url` is for the `main` (default) branch:

```
web@app.0:~$ echo $PLATFORM_ROUTES | base64 --decode | jq
{
  "https://test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/": {
    "primary": true,
    "id": null,
    "production_url": "https://main-bvxea6i-ev7iougo4jhve.us-2.platformsh.site/",
    "attributes": {},
    "type": "upstream",
    "upstream": "app",
    "original_url": "https://{default}/"
  },
  "https://www.test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/": {
    "primary": false,
    "id": null,
    "production_url": "https://www.main-bvxea6i-ev7iougo4jhve.us-2.platformsh.site/",
    "attributes": {},
    "type": "redirect",
    "to": "https://test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/",
    "original_url": "https://www.{default}/"
  },
  "http://test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/": {
    "original_url": "http://{default}/",
    "id": null,
    "primary": false,
    "type": "redirect",
    "to": "https://test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/",
    "production_url": "http://main-bvxea6i-ev7iougo4jhve.us-2.platformsh.site/"
  },
  "http://www.test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/": {
    "original_url": "http://www.{default}/",
    "id": null,
    "primary": false,
    "type": "redirect",
    "to": "https://www.test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/",
    "production_url": "http://www.main-bvxea6i-ev7iougo4jhve.us-2.platformsh.site/"
  }
}
```

This prompted me to look at how the platform `config-reader` library builds these routes. It seems we should be looking for all `type = upstream` routes. This is well documented in the source here! :-) https://github.com/platformsh/config-reader-php/blob/09982f28b0831401dc62a34fa3edceda76efaf19/src/Config.php#L325-L348

This makes the JQ a bit more complex because the URLs we want to target are JSON object keys, but I can replicate the same the same logic with this:

```
web@app.0:~$ echo $PLATFORM_ROUTES | base64 --decode | jq -r 'to_entries | map(select(.value.type == "upstream")) | sort_by(.value.primary, .key) | reverse | [.[].key] | first'
https://test-t6dnbai-ev7iougo4jhve.us-2.platformsh.site/
```

Reply:
I've added a commit to implement this logic

Reply:
@paul121 Impressive sleuthing, and also excellent documentation of your thinking and discovery process. I think you found a great path forward.